### PR TITLE
Add Spring Security OAuth

### PR DESCRIPTION
**Pull request by SWhitlam**
Integrate with HMPPS Auth
Authorise endpoints with Role and Scope
Setup Docker Compose
Add Controller integration tests for Auth

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Test
+    fun `access forbidden when no read scope`() {
+      webTestClient.post().uri("/risks/supplementary")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf()))
+        .bodyValue(requestBody)
+        .exchange()
+        .expectStatus().isForbidden
```
Comment (LucasCairns) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt:163]: Do we have a happy path for `PostNewSupplementaryRisk`?

**Review by LucasCairns — DISMISSED** [review]
LGTM 👍

**Review by adelaidaa — APPROVED** [review]
Great work Guys!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt (hunk 1 above)
Comment (SWhitlam) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt:163]: Just added it

Comment (LucasCairns) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt:163]: 👍

**Review by LucasCairns — APPROVED** [review]
👍

**Pull request merged**